server/src/main/java/org/elasticsearch/common/breaker/ChildMemoryCircuitBreaker.java

```diff
@@ -57,17 +58,24 @@ public ChildMemoryCircuitBreaker(BreakerSettings settings, Logger logger, Circui
      * the given overheadConstant. Uses the given oldBreaker to initialize
      * the starting offset.
      * @param settings settings to configure this breaker
-     * @param parent parent circuit breaker service to delegate tripped breakers to
-     * @param name the name of the breaker
      * @param oldBreaker the previous circuit breaker to inherit the used value from (starting offset)
+     * @param logger the logger instance to use
+     * @param parent parent circuit breaker service to delegate tripped breakers to
      */
     public ChildMemoryCircuitBreaker(BreakerSettings settings,
                                      ChildMemoryCircuitBreaker oldBreaker,
                                      Logger logger,
                                      CircuitBreakerService parent) {
         this.name = settings.getName();
-        this.settings = settings;
         this.memoryBytesLimit = settings.getLimit();
+        if (memoryBytesLimit == -1) {
+            freeSupplier = () -> Long.MAX_VALUE;
+        } else if (memoryBytesLimit == 0) {
+            freeSupplier = () -> 0L;
+        } else {
+            freeSupplier = () -> getLimit() - getUsed();
+        }
```

Comment:
Could it also be set to other negative values, or is that forbidden already?

Reply:
This is forbidden already, see https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java#L43